src/routes/blog/post/remix-3-whats-changing-and-why-it-matters/+page.markdoc

```diff
@@ -0,0 +1,144 @@
+---
+layout: post
+title: "Remix 3: what's changing and why it matters"
+description: "A developer-focused deep dive into Remix 3's principles, architecture, and why it moves beyond React toward web standards."
+date: 2025-11-12
+cover: /images/blog/remix-3-whats-changing-and-why-it-matters/cover.png
+timeToRead: 8
+author: ebenezer-don
+category: news
+---
+
+For a while, it seemed like Remix had reached a natural end. By late 2024, almost everything that made Remix unique, its loaders, actions, and nested routing, had been merged into React Router 7. The framework's creators, Ryan Florence and Michael Jackson, even joked that Remix 2 had become "React Router, the Framework." Many developers assumed Remix would fade away.
+
+But in May 2025, the team announced Remix 3, an unexpected rebuild from the ground up, meant to rethink how web apps are built. They showed early previews at Remix Jam 2025, and while the final release is planned for 2026, the goals are already clear. Remix 3 breaks away from React entirely and rebuilds around web standards.
+
+It's a big step. The framework now runs directly on the Fetch API, not Node's request/response system. It can run on any JavaScript runtime; Node, Deno, Bun, or Cloudflare Workers, without adapters. It still supports the same development patterns developers liked in Remix 2, but without depending on React's lifecycle or its virtual DOM. The team calls this a move toward being "closer to the web itself."
```

Comment:
Cloudflare Workers is direct competitor of Functions, not sure if we should mention them. And if yes, we might want to mention Appwrite Functions too